athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/DynamoDBRecordHandler.java

```diff
@@ -219,6 +221,12 @@ private AmazonWebServiceRequest buildReadRequest(Split split, String tableName)
             }
         }
 
+        // Only read columns that are needed in the query
+        String projectionExpression = schema.getFields()
+                .stream()
+                .map(field -> field.getName())
```

Comment:
do we need to alias columns in case they are in the filter expression?

Reply:
schema already has the dynamodb column names (after columnNameMapping is applied) so no need to do more

Reply:
I meant the '#' prepending for filter expressions

Reply:
that is taken care of in the filterExpression and already contains the right format

Reply:
I vaguely remember having to do it for the projection expression as well but if you tested this with `select a from table where <constraint on a>` then I'm good.

Reply:
yep one of the tests ive done is 

select i.item_id, i.cost, o.ship_address.city from itemtable i, ordertable o where i.item_id = o.item_id and o.returned = false order by i.item_name;

Reply:
can you try `select o.returned, i.item_id, i.cost, o.ship_address.city from itemtable i, ordertable o where i.item_id = o.item_id and o.returned = false order by i.item_name;`?

the item_id constraint isn't pushed down

Reply:
actually 'returned' is in both the projection and filter so we should be good.